packages/lexical/src/LexicalNode.ts

```diff
@@ -185,6 +188,46 @@ export class LexicalNode {
   __prev: null | NodeKey;
   /** @internal */
   __next: null | NodeKey;
+  /**
+   * Don't use this directly, use `this.getClasses()` and `this.mutateClasses()` instead
+   * @internal
+   */
+  __classes?: ReadOnlyClasses;
+
+  /**
+   * Returns an object of classes in the form of `prefix-suffix` for string values, or just `prefix` for true boolean values.
+   * @example
+   * const exampleClassesObject = {
+   *   bg: 'red', // the node is rendered with class `bg-red`
+   *   text: 'green', // node is rendered with class `text-green`,
+   *   active: true, // node is rendered with class `active`,
+   * }
+   * // Resulting classes: 'bg-red', 'text-green', and 'active'
+   *
+   * @returns The classes object.
+   */
+  getClasses(): ReadOnlyClasses {
+    const self = this.getLatest();
+    return self.__classes || {};
+  }
+
+  /**
+   * Allows mutation of the classes object where the key-value pairs follow the format `prefix-suffix` for string values,
+   * or just `prefix` for true boolean values.
+   *
+   * @example
+   * node.mutateClasses((currentClasses) => {
+   *   currentClasses.bg = 'blue'; // the node will now be rendered with class `bg-blue`
+   *   delete currentClasses.active; // the node will no longer have the class `active`
+   * });
+   *
+   * @param fn A function that receives the current classes object and allows it to be mutated safely.
+   */
+  mutateClasses(fn: (classes: MutableClasses) => void) {
+    const self = this.getWritable();
+    self.__classes = self.__classes || {};
+    fn(self.__classes);
+  }
```

Comment:
This is what the sort of copy on write pattern I suggested would look like:

```suggestion
  mutateClasses(fn: (classes: MutableClasses) => void | ReadonlyClasses) {
    const self = this.getWritable();
    const classes = {...self.__classes};
    self.__classes = fn(classes) || classes;
  }
```

All other copies can go away since a readonly object doesn't need to be copied. The comparisons can also be short-circuited when the objects are identical which would be the case unless a mutation happened.